ncm-network/src/main/pan/components/network/types/network/interface.pan

```diff
@@ -34,13 +34,25 @@ type network_interface_alias = {
     Describes the bonding options for configuring channel bonding on EL5 and similar.
 }
 type network_bonding_options = {
-    "mode" : long(0..6)
-    "miimon" : long
-    "updelay" ? long
-    "downdelay" ? long
-    "primary" ? valid_interface
-    "lacp_rate" ? long(0..1)
-    "xmit_hash_policy" ? choice('0', '1', '2', 'layer2', 'layer2+3', 'layer3+4')
+    "ad_select"         ? choice('bandwidth', 'count', 'stable')
+    "all_slaves_active" ? long(0..1)
+    "arp_interval"      ? long
+    "arp_ip_target"     ? type_ip[]
+    "arp_validate"      ? choice('none', 'active', 'backup', 'all')
+    "downdelay"         ? long
+    "fail_over_mac"     ? choice('active', 'follow', 'none')
+    "lacp_rate"         ? long(0..1)
+    "min_links"         ? long
+    "miimon"            : long
```

Comment:
Switch to optional? The [doc](https://docs.rs/nmstate/latest/nmstate/struct.BondOptions.html#fields) suggests default is 0 (i.e. off) -- is that more sensible than forcing a user to specify a value?

Reply:
I havent  changed miimon, this was already there in current release.  happy to change it to optional if agreed.

Reply:
If it makes sense either change it to optional or define a default value so that there is no requirement to add it if you don't need to.